scripts/get_garage_sign.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+import argparse
+import hashlib
+import os.path
+import requests
+import shutil
+import sys
+import tarfile
+import xml.etree.ElementTree as ET
+
+from pathlib import Path
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Download a specific or the latest version of garage-sign')
+    parser.add_argument('-n', '--name', help='specific version to download')
+    parser.add_argument('-s', '--sha256', help='expected hash of requested version')
+    parser.add_argument('-o', '--output', type=Path, default=Path('.'), help='download directory')
+    args = parser.parse_args()
+    if not args.output.exists():
+        print('Error: specified output directory ' + args.output + ' does not exist!')
+        return 1
+    sha256_hash = args.sha256
+    if sha256_hash and not args.name:
+        print('Warning: sha256 hash specified without specifying a version.')
+    if args.name and not sha256_hash:
+        print('Warning: specific version requested without specifying the sha256 hash.')
+
+    r = requests.get('https://ats-tuf-cli-releases.s3-eu-central-1.amazonaws.com')
+    if r.status_code != 200:
+        print('Error: unable to request index!')
+        return 1
+    tree = ET.fromstring(r.text)
+    # Assume namespace.
+    ns = '{http://s3.amazonaws.com/doc/2006-03-01/}'
+    items = tree.findall(ns + 'Contents')
+    versions = dict()
```

Comment:
nitpick but perhaps too many list comprehension:

```
for i in items:
    if not i.find(ns + 'Key').text.startswith('cli-'):
        continue
    ...
```

Or use an intermerdiary variable

```
cli_items = [i for i in items if i.find(ns + 'Key').startswith('cli-')]
for i in cli_items:
    ...
```

Reply:
They should be equivalent functionally except if I made a fatal mistake, this is only about readability, which is often subjective. I tested the first version by running the program without argument and it successfully downloaded the archive.

Other than that, it's only a matter of style, really... The first version looks more like traditional imperative programming.